app/assets/javascripts/3dbio_viewer/src/data/PdbPublications.ts

```diff
@@ -0,0 +1,86 @@
+import { Codec, GetType, array, nullType, number, string } from "purify-ts";
+import { maybe } from "./PdbLigands";
+import { PdbPublication } from "../domain/entities/Pdb";
+
+export const publicationsCodec = array(
+    //using so many "maybe" from examples: 7O7Z, 6ABA, 7SUB
+    //but truly I don't want to use them so often. Check with JR
+    //9zzz will return {}
+    Codec.interface({
+        doi: maybe(string),
+        title: string,
+        pubmed_id: maybe(string),
+        type: string,
+        associated_entries: maybe(string),
+        journal_info: Codec.interface({
+            pdb_abbreviation: maybe(string),
+            ISO_abbreviation: maybe(string),
+            pages: maybe(string),
+            volume: maybe(string),
+            issue: maybe(string),
+            year: maybe(number),
+        }),
+        abstract: Codec.interface({
+            background: nullType,
+            objective: nullType,
+            methods: nullType,
+            results: nullType,
+            conclusions: nullType,
+            unassigned: maybe(string),
+        }),
+        author_list: array(
+            Codec.interface({
+                full_name: string,
+                last_name: maybe(string),
+                initials: maybe(string),
+            })
+        ),
+    })
+);
+
+export function getPublicationsCodec(pdbId: string): Codec<EntryPublications> {
+    return Codec.interface({
+        [pdbId]: publicationsCodec,
+    });
+}
+
+export type EntryPublications = {
+    [x: string]: PublicationsCodec;
+};
```

Comment:
This way of writing a record type allows to be specific about the nature of the key. Let's use a declarative name for this `x`.